### PR TITLE
TensorFlow estimator example with only v2 APIs

### PR DESCRIPTION
**Pull request by nuka137**
This PR adds Tensorflow estimator example with v2 API.
Against #868 uses many `tf.compat.v1` which will be deprecated in the future, this new example only uses v2 API.
TensorFlow estimator now supports lots of premade estimators, so I think using these estimators are good start points.
https://www.tensorflow.org/guide/estimator

This example uses DNNClassifier (https://www.tensorflow.org/api_docs/python/tf/estimator/DNNClassifier) which reproduces old example as possible.

Related Issue: #883

### CODE REVIEW

File: examples/tensorflow_estimator_simple.py [hunk 1]
```diff
@@ -53,68 +53,49 @@ def create_optimizer(trial):
...
-    train_input_fn = tf.compat.v1.estimator.inputs.numpy_input_fn(
-        x={"x": train_data}, y=train_labels, batch_size=BATCH_SIZE, num_epochs=None, shuffle=True)
+    classifier = create_classifier(trial)
 
-    mnist_classifier.train(input_fn=train_input_fn, steps=TRAIN_STEPS)
+    classifier.train(input_fn=input_fn, steps=TRAIN_STEPS)
 
-    eval_input_fn = tf.compat.v1.estimator.inputs.numpy_input_fn(
-        x={"x": eval_data}, y=eval_labels, num_epochs=1, shuffle=False)
+    eval_results = classifier.evaluate(input_fn=input_fn, steps=1)
```
Comment (toshihikoyanase) [inline examples/tensorflow_estimator_simple.py]: This code uses the training accuracy as the objective value while the previous code uses test accuracy. Please use the test data to avoid overfitting to the training data.

In addition, I think this code use only `BATCH_SIZE` images in the dataset because `steps` is set to `1` (c.f., [DNNClassifier.evaluate](https://www.tensorflow.org/api_docs/python/tf/estimator/DNNClassifier#evaluate)).  How about removing the `steps` argument and use the whole dataset? In that case, we can restrict the number of images using [`take()`](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#take) method of `tfds.Dataset` as can be seen in https://github.com/optuna/optuna/blob/master/examples/tensorflow_eager_simple.py#L105.

File: examples/tensorflow_estimator_simple.py [hunk 2]
```diff
@@ -53,68 +53,49 @@ def create_optimizer(trial):
     optimizer_name = trial.suggest_categorical('optimizer', ['Adam', 'SGD'])
     if optimizer_name == 'Adam':
         adam_lr = trial.suggest_loguniform('adam_lr', 1e-5, 1e-1)
-        optimizer = tf.compat.v1.train.AdamOptimizer(learning_rate=adam_lr)
+        def optimizer(): return tf.keras.optimizers.Adam(learning_rate=adam_lr)
```
[the remark this answers is not in the document]
Comment (nuka137) [reply examples/tensorflow_estimator_simple.py]: It's a good idea, so I used `lambda` instead.

File: examples/tensorflow_estimator_simple.py (hunk 1 above)
Comment (nuka137) [reply examples/tensorflow_estimator_simple.py]: Thanks! I fixed them.

**Comment by nuka137** [comment]
@toshihikoyanase 

Thanks for your fast review!
Now, I fixed all issues you mentioned.

And, I'm so sorry to forget updating to v2 API about `examples/pruning/tensorflow_estimator_integration.py` .
I will check it and make new PR, ASAP.
